.github/workflows/sdk-execution.yml

```diff
@@ -0,0 +1,74 @@
+name: KFP SDK execution tests
+
+on:
+  push:
+    branches: [master]
+
+  pull_request:
+    paths:
+      - '.github/workflows/sdk-execution.yml'
+      - 'sdk/python/**'
+      - 'api/v2alpha1/**'
+
+jobs:
+  sdk-execution-tests:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Create k8s Kind Cluster
+        uses: container-tools/kind-action@v2
+        with:
+          cluster_name: kfp
+          kubectl_version: v1.29.2
+          version: v0.22.0
+          node_image: kindest/node:v1.29.2
+
+      - name: Build images
+        run: ./scripts/deploy/github/build-images.sh
+
+      - name: Deploy KFP
+        run: ./scripts/deploy/github/deploy-kfp.sh
+
+      - name: Forward API port
+        run: ./scripts/deploy/github/forward-port.sh "kubeflow" "ml-pipeline" 8888 8888
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
```

Comment:
Don't we want to test with the other versions as well?
You can use a matrix for that. See https://github.com/kubeflow/pipelines/pull/10926/files#diff-e842340381d5ec7517ff5c2522b081e6b081cad5701ce9f06af7d088428974eeR17-R24

Reply:
the Prow Job used only Python 3.8: https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-presubmits.yaml#L238

Do we need this for other versions? I would suggest keeping it simple, to one version, and not adding more jobs / more time to execute tests for every PR.

Reply:
> Do we need this for other versions?
I don't have a strong opinion, but testing with all supported versions would be nice.

> I would suggest keeping it simple, to one version, and not adding more jobs / more time to execute tests for every PR.
The jobs run in parallel, so unless we reach the limit of parallel executions, it shouldn't increase the execution time.

@rimolive any thoughts?

Reply:
Only 4 jobs run in parallel. I just saw that on my PR.
So yeah, it would increase the execution time.

![Screenshot from 2024-07-02 13-22-38](https://github.com/kubeflow/pipelines/assets/11776454/4a98a466-d584-4b65-9a41-b335db564cfe)